providers/hns/hns_roce_u_hw_v2.c

```diff
@@ -1091,6 +1091,9 @@ static int hns_roce_u_v2_modify_qp(struct ibv_qp *qp, struct ibv_qp_attr *attr,
 	struct hns_roce_qp *hr_qp = to_hr_qp(qp);
 	bool flag = false; /* modify qp to error */
 
+	if (attr_mask & IBV_QP_STATE)
```

Comment:
No way, that updating QP state before calling to modify_qp  is correct.

Reply:
Thank you. I move it after modify_qp.